can/io/blf.py

```diff
@@ -583,11 +583,12 @@ def file_size(self) -> int:
     def stop(self):
         """Stops logging and closes the file."""
         self._flush()
-        if self.file.seekable():
-            filesize = self.file.tell()
-            # Write header in the beginning of the file
-            self.file.seek(0)
-            self._write_header(filesize)
```

Comment:
@zariiii9003 The call `self.file.seek(0)` results in an OSError. It appears that the code block is not required if write mode `wb` is used (rather than `ab`). However, if the `append` option is requested, then some of this idea may be required.

Reply:
The header contains things that are only known after the log has been written, so that's why we need to go back and write it later. Previously we closed and reopened the file to update it, but it seemed unnecessary if it was possible to seek instead.